### ISSUE TITLE
[JENKINS-16552] Throttle plugin does not throttle sub jobs in matrix project

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/matrix-project-plugin#533]
<p>Throttling is only applied on parent job. </p>

<p>A shared and limited resource (e.g. the number of licences) needs to be used by several child jobs, split over several matrix jobs. Not all parts of the matrix job can be executed in parallel, but running in sequence is neither an option.</p>

<p>The Throttle-concurrent addon do this kind of throttling, however it is only applied at the master job and not at the individual sub jobs, so instead of running 5-10 instances of the massively parallel job all 500 variants are queued up at once blocking other kinds of jobs from running.</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=simson">simson</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-16552" target="_blank">Throttle plugin does not throttle sub jobs in matrix project</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>component(s)</b>: matrix-project-plugin, throttle-concurrent-builds-plugin
<li><b>resolution</b>: Duplicate
<li><b>resolved</b>: 2013-02-01T10:43:37+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 2
<li><b>imported</b>: 2025-12-06
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Throttling is only applied on parent job. </p>

<p>A shared and limited resource (e.g. the number of licences) needs to be used by several child jobs, split over several matrix jobs. Not all parts of the matrix job can be executed in parallel, but running in sequence is neither an option.</p>

<p>The Throttle-concurrent addon do this kind of throttling, however it is only applied at the master job and not at the individual sub jobs, so instead of running 5-10 instances of the massively parallel job all 500 variants are queued up at once blocking other kinds of jobs from running.</p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>cjw296:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/16552?focusedId=173070&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-173070">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-16552?focusedId=173070&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-173070">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This is a duplicate of:</p>

<p><a href="https://issues.jenkins-ci.org/browse/JENKINS-16521" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-16521</a></p>

<p>...where I've accumulated more info and references to other related issues.</p>
</pre>
</details>

</ul>
</details>

<p>This is a duplicate of:</p>

<p><a href="https://issues.jenkins-ci.org/browse/JENKINS-16521" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-16521</a></p>

<p>...where I've accumulated more info and references to other related issues.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>simson:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/16552?focusedId=173365&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-173365">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-16552?focusedId=173365&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-173365">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Close - duplicate</p>
</pre>
</details>

</ul>
</details>

<p>Close - duplicate</p>
